pkg/cmd/run.go

```diff
@@ -136,6 +138,25 @@ doppler run --mount secrets.json -- cat secrets.json`,
 		// retrieve secrets
 		dopplerSecrets := fetchSecrets(localConfig, enableCache, fallbackOpts, metadataPath, nameTransformer, dynamicSecretsTTL)
 
+
+		if cmd.Flags().Changed("only-secrets") {
+			if len(secretsToInclude) == 0 {
+				utils.HandleError(fmt.Errorf("you cannot use --only-secrets with no secrets"))
+			}
+
+			var err error
+			noExitOnMissingIncludedSecrets := cmd.Flags().Changed("no-exit-on-missing-included-secrets")
+			dopplerSecrets, err = controllers.SelectSecrets(dopplerSecrets, secretsToInclude)
```

Comment:
I'm not sure that this controller function is providing much value. I think we could just call `filter` directly here.

To detect if all the specified secrets exist, you can compare the length of the return value against the length of `secretsToInclude`.

Reply:
I think this is valuable because offers a function to select secrets provided you give the original map and the secrets that you want to be included and potentially an error (some secrets the user wanted to include do not exist in the original secret map)

Behind the scenes, the `SelectSecrets` calls the generic `FilterMap` function which returns a new secret map with the secrets that were selected as the first return value and a second return value that is either an array of keys that were did not exist in the original map or nil it didn't find any non existant secrets.

Reply:
Added a note in the other thread which I think would give this controller function some purpose. Let's consolidate these threads and continue the discussion there.
https://github.com/DopplerHQ/cli/pull/341#discussion_r1014312392